[18.0][MIG] account_invoice_show_currency_rate

JordiMForgeFlow opened the pull request:
Migration to V18:

- Odoo adds a new field to display the currency rate on invoices when the currency is different than the company's currency (https://github.com/odoo/odoo/blob/e3d54695862f73d9be6c35121def57ed641a1a45/addons/account/models/account_move.py#L462). However, the computed rate is simply based on the rate configured in the system at the time of computing. It could be the case that a different rate was existing when the invoice was posted, so this module ensures that the rate computed for posted invoices is based on the actual currency amount applied.

FW of https://github.com/OCA/account-invoicing/pull/1838:

- Odoo computes the currency rate on journal items (https://github.com/odoo/odoo/blob/e3d54695862f73d9be6c35121def57ed641a1a45/addons/account/models/account_move_line.py#L133). However, the computed rate is based on the rate configured in the system. It could be the case for manually created entries that the user has manually modified the amount in currency, so this module ensures that the computed rate is based on the actual currency amount applied.

FW of https://github.com/OCA/account-invoicing/pull/1899 and https://github.com/OCA/account-invoicing/pull/1901

victoralmau committed:
[ADD] account_invoice_show_currency_rate: Show currency rate in invoices

oca-travis committed:
[UPD] Update account_invoice_show_currency_rate.pot

OCA-git-bot committed:
[UPD] README.rst

victoralmau committed:
[IMP] account_invoice_show_currency_rate: Make sure the currency is correct in the main company

OCA-git-bot committed:
account_invoice_show_currency_rate 13.0.1.0.1

Commit:
[FIX] account_invoice_show_currency_rate: Use assertAlmostEqual in tests

victoralmau committed:
[FIX] account_invoice_show_currency_rate: Round currency_rate_amount to avoid many decimal places (1.9974358974358974 for example).

OCA-git-bot committed:
account_invoice_show_currency_rate 13.0.1.0.2

Commit:
[IMP] account_invoice_show_currency_rate: pre-commit stuff

Commit:
[MIG] account_invoice_show_currency_rate: Migration to 16.0

Commit:
[UPD] Update account_invoice_show_currency_rate.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
account_invoice_show_currency_rate 16.0.1.0.1

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_show_currency_rate
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_show_currency_rate/

Gelo-fl committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (3 of 3 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_show_currency_rate
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_show_currency_rate/es/

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_show_currency_rate
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_show_currency_rate/it/

JordiMForgeFlow committed:
[IMP] account_invoice_show_currency_rate: optional show journal items rate

rousseldenis committed:
[FIX] account_invoice_show_currency_rate: Use TransactionCase for tests

Commit:
[UPD] Update account_invoice_show_currency_rate.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_show_currency_rate
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_show_currency_rate/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_show_currency_rate
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_show_currency_rate/it/

OCA-git-bot committed:
[BOT] post-merge updates

jeer2234 committed:
[IMP] : pre-commit auto fixes

jeer2234 committed:
[MIG] account_invoice_show_currency_rate: Migration to 17.0

JordiMForgeFlow committed:
[FIX] account_invoice_show_currency_rate: use only debit from currency lines to get rate

OCA-git-bot committed:
[BOT] post-merge updates

JordiMForgeFlow committed:
[IMP] account_invoice_show_currency_rate: add currency rate in journal items

JordiMForgeFlow committed:
[IMP] account_invoice_show_currency_rate: use absolute balance to get rate on credit amounts

JordiMForgeFlow committed:
[MIG] account_invoice_show_currency_rate: Migration to V18

GuillemCForgeFlow reviewed (approved):
makes sense to me 👍🏿

AaronHForgeFlow reviewed (approved):
Functional + code

JasminSForgeFlow reviewed (approved):
Code review

JordiMForgeFlow committed:
[IMP] account_invoice_show_currency_rate: prevent error when balance is zero

JordiMForgeFlow committed:
[FIX] account_invoice_show_currency_rate: do not round currency rate

Currency rates are not rounded in Odoo. Applying the rounding causes the
currency rate of the line to be different when the move is posted.

AlexPForgeFlow reviewed (approved):
LGTM!

JordiBForgeFlow commented:
/ocabot merge

JordiBForgeFlow commented:
/ocabot merge nobump
